### PR TITLE
[12.0] MIG purchase_commercial_partner

### PR DESCRIPTION
**Pull request by alexis-via**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please squash together adjacent "OCA Transbot.." commits using https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

File: purchase_commercial_partner/models/purchase.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+from odoo import fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    commercial_partner_id = fields.Many2one(
+        'res.partner', related='partner_id.commercial_partner_id', store=True,
+        index=True, string='Commercial Vendor', readonly=True)
```
Comment (nikul-serpentcs) [inline purchase_commercial_partner/models/purchase.py]: @alexis-via no need to `readonly`

Comment (bealdav) [reply purchase_commercial_partner/models/purchase.py]: I think yes, related fields with full access leads to weird and unwanted behavior, bad performance: crazy sql.

In master, related fields are now readonly by default, isn't it ?

Comment (pedrobaeza) [reply purchase_commercial_partner/models/purchase.py]: In v12, related are already `readonly` by default.

Comment (bealdav) [reply purchase_commercial_partner/models/purchase.py]: you're right ;-)
https://github.com/odoo/odoo/blob/12.0/odoo/fields.py#L422

Comment (pedrobaeza) [reply purchase_commercial_partner/models/purchase.py]: We included in migration guide the reverse: if you rely on default `False` value (for example, for res.config.settings fields), you have to explicitly set the contrary: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0#tasks-to-do-in-the-migration

That's why the Python mantra "Explicit better than implicit" works. For this case, it's harmless to leave it explicit.

**Review by HviorForgeFlow — APPROVED** [review]
Code review LGTM

**Review by bealdav — APPROVED** [review]

**Pull request merged**